assets/src/apps/authoring/components/PropertyEditor/schemas/screen.ts

```diff
@@ -116,16 +117,48 @@ export const screenUiSchema = {
   },
 };
 
-export const getScreenData = (data: any) => {
+export const transformScreenModeltoSchema = (data: any) => {
   if (data) {
+    const schemaPalette = {
+      ...data.palette,
+      borderWidth: `${data.palette.lineThickness ? data.palette.lineThickness + 'px' : '1px'
+        }`,
+      borderRadius: '10px',
+      borderStyle: 'solid',
+      borderColor: `rgba(${data.palette.lineColor || data.palette.lineColor === 0
+        ? chroma(data.palette.lineColor).rgb().join(',')
+        : '255, 255, 255'
+        },${data.palette.lineAlpha})`,
+      backgroundColor: `rgba(${data.palette.fillColor || data.palette.fillColor === 0
+        ? chroma(data.palette.fillColor).rgb().join(',')
+        : '255, 255, 255'
+        },${data.palette.fillAlpha})`
+    }
     return {
       ...data,
-      useHtmlProps: data.palette.useHtmlProps,
       Size: { width: data.width, height: data.height },
       checkButton: { showCheckBtn: data.showCheckBtn, checkButtonLabel: data.checkButtonLabel },
       max: { maxAttempt: data.maxAttempt, maxScore: data.maxScore },
+      palette: schemaPalette
```

Comment:
good but we only want to override these if useHtmlProps is falsey. that would mean that it hasn't yet come into this editor before (because we're going to flip it to true on the way out). So, we only want to convert the values if that is the case, because otherwise it won't retain the settings they've chosen for backgroundColor directly.